edge-modules/ModuleRestarter/src/Program.cs

```diff
@@ -0,0 +1,62 @@
+// Copyright (c) Microsoft. All rights reserved.
+namespace ModuleRestarter
+{
+    using System;
+    using System.Threading;
+    using System.Threading.Tasks;
+    using Microsoft.Azure.Devices;
+    using Microsoft.Azure.Devices.Edge.Util;
+    using Microsoft.Extensions.Logging;
+
+    class Program
+    {
+        static readonly ILogger Log = Logger.Factory.CreateLogger<Program>();
+        static readonly string ServiceClientConnectionString = Settings.Current.ServiceClientConnectionString;
+        static readonly string DeviceId = Settings.Current.DeviceId;
+        static readonly string DesiredModulesToRestartCSV = Settings.Current.DesiredModulesToRestartCSV;
+        static readonly int RandomRestartIntervalInMins = Settings.Current.RandomRestartIntervalInMins;
+
+        public static int Main() => MainAsync().Result;
+
+        static async Task<int> MainAsync()
+        {
+            Log.LogInformation($"Starting module restarter with the following settings:\r\n{Settings.Current}");
+
+            if (string.IsNullOrEmpty(DesiredModulesToRestartCSV))
+            {
+                Log.LogInformation("No modules specified to restart. Stopping.");
+                return 0;
+            }
+
+            (CancellationTokenSource cts, ManualResetEventSlim completed, Option<object> handler) = ShutdownHandler.Init(TimeSpan.FromMinutes(2 * RandomRestartIntervalInMins), null);
+
+            await RestartModules(cts, RandomRestartIntervalInMins);
+            completed.Set();
+

```

Comment:
you need to call below method for cleanup, please check out other modules, e.g. DirectMethodCloudSender.
```suggestion
handler.ForEach(h => GC.KeepAlive(h));
```